server/controllers/commentController.js

```diff
@@ -0,0 +1,208 @@
+        comments
+      });
+
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * @param {object} req - req from route
+   * @param {object} res - respose to route
+   * @param {object} next - callback function
+   * @returns {object} a response object
+   */
+  static async getSingleComment(req, res, next) {
+    const { params: { id }} = req;
+    try {
+      const comment = await Comment.findOne(
+        {
+          include: [{
+            model: User,
+            as: 'author',
+            attributes: ['userName', 'imageUrl', 'bio']
+
+          }],
+          attributes: { exclude: ['userId'] },
+          where: { id }
+        });
+
+      if (!comment) {
+        return res.status(404).json({
+          success: false,
+          message: 'No Comment found',
+        });
+      }
+
+      return res.status(200).json({
+        success: true,
+        message: 'Retrieved comment successfully',
+        comment
+      });
+
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * @description only owners of comment should be able to update
+   * @param {object} req - req from route
+   * @param {object} res - respose to route
+   * @param {object} next - callback function
+   * @returns {object} a response object
+   */
+  static async updateComment(req, res, next) {
+    const { user:{ id },body:{ comment }, params:{ commentId }} = req;
+    try {
+      const checkCommentOwner = await Comment.findOne({
+        where: {
+          id: commentId,
+          userId: id,
+         }
+      });
+      if (!checkCommentOwner) {
+        return res.status(403).json({
+          success: false,
+          message: 'User cannot update comment they did\'t create',
+        });
+      }
+      await Comment.update(
+        {
+          comment
+        },
+        {
+          where: { id: commentId }
+        }
+      );
+      return res.status(200).json({
+        success: true,
+        message: 'Comment updated successfully',
+      });
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+   /**
+   * @description A function that deletes comments from the database
+   * @param {uuid} commentId - req from route
+   * @param {uuid} userId - respose to route
+   * @returns {boolean} a response object
+   */
+  static async deleteCommentFunction(commentId,userId){
+    const comment = await Comment.findOne({
+      where: {
+        id: commentId,
+        userId,
+      }
+    });
+    if (!comment) {
+      return false;
+
+    }
+    await comment.destroy({
+      where: {
+        id:commentId
+      }
+    });
+    return true;
+  }
+
+  /**
+   * @description Users should be able to delete only their comment
+   * @param {object} req - req from route
+   * @param {object} res - respose to route
+   * @param {object} next - callback function
+   * @returns {object} a response object
+   */
+  static async deleteComment(req, res, next) {
```

Comment:
You do not need two functions for delete operation. One function should handle ascertaining who is able to delete a comment while the other should handle the delete process.